database/postgres/postgres.go

```diff
@@ -131,7 +131,7 @@ func (p *Postgres) Lock() error {
 
 	// This will either obtain the lock immediately and return true,
 	// or return false if the lock cannot be acquired immediately.
-	query := `SELECT pg_try_advisory_lock($1)`
+	query := `SELECT pg_advisory_lock($1)`
 	var success bool
 	if err := p.conn.QueryRowContext(context.Background(), query, aid).Scan(&success); err != nil {
```

Comment:
Wanna try using `context.WithTimeout()` instead of implementing your own timeout?

Reply:
so I was thinking about this, but the driver has no idea of what the configured locked timeout is, it isnt passed down and a context isnt passed down either.  It would be best if the interface was changed to pass a context down to the lock/unlock methods.  Then a timeout could be set on that context.